### ISSUE TITLE
Adding blog posts

### ISSUE THREAD

**Issue by SimonAlling** [lydell/blog#3]
**TL;DR** If you can update a blog really easily, you’ve won. Name your posts `#1`, `#2` and `#3-whatever`.

## The problem

Let’s say you’re authoring some blog that has a lot of posts in it. In particular, this post: #2. Every time you read that post, you simply read `color: #42aad5;`, `background-color: #42aad5;`, `box-shadow: 0 0 5px #42aad5;` or whatever. So far so good.

Then one day, a reader says to you: “You know that blog with color codes all over the place? Can you add a post to it, so that it ‘pops’ a little more?”

Now you have plenty of changes to make. Easy enough, though. Just create a new post with ‘popping’ content.

But shortly after your change, the reader comes to you again. “The new post does not seem to show up on the blog, could you take a look at what’s going on?”

Turns out the path had been written uppercase in the database for some reason: `/post/POPPING`. Better remember to do a case insensitve path routing next time.

But it doesn’t end there. If Unicode has been involved during the session, ever-so-slightly-off variations of the path might exist. Perhaps your reader wrote `pöpping` instead of `popping`. Nobody’s eyes can tell the difference, but it’s enough to fool your routing.

## A solution

Use a VCS blog platform. I like [GitHub](/). You can also use [GitLab](https://gitlab.com), [SourceForge](https://sourceforge.net), [Bitbucket](https://bitbucket.org) with some plugin, or a custom repository on a modern server, but I’ll go with GitHub in this post.

In lots of blogs I’ve worked on, there’s an [issues](/lydell/blog/issues) page containing global posts for, well, global stuff. `package.json` files, npm warnings, some Star Wars stuff–and, most importantly, naming CSS colors.

With issues for posts, the popping update would have been super easy:

[`new`](new)

Teach your readers to look for posts in issues instead of typing in their paths and you should be fine.

## Conclusion

Blogs are important. Make them easy to use. Easy to change. Easy to talk about with other people. Don’t make blogging hard for yourself. Be practical. And allow a little fun in your blog. :)
